lib/mutant/reporter/hash/printer.rb

```diff
@@ -0,0 +1,52 @@
+module Mutant
+  class Reporter
+    class Hash
+      # Hash runner status printer base class
+      class Printer
```

Comment:
This base class duplicates a lot with `Mutant::Reporter::CLI::Printer`, I suggest deduplicating that into `Mutant::Reporter::Printer`.